src/amo/components/RatingsByStar/index.js

```diff
@@ -86,85 +91,100 @@ export class RatingsByStarBase extends React.Component<InternalProps> {
     );
   }
 
+  handleRouting(star: string) {
+    const { addon, location, history, clientApp, siteLang } = this.props;
+
+    invariant(addon, 'addon was unexpectedly empty');
+
+    history.push(
+      `/${siteLang}/${clientApp}${reviewListURL({
+        addonSlug: addon.slug,
+        score: star,
+        src: location.query.src,
+      })}`,
+    );
+  }
+
   render() {
-    const { addon, errorHandler, i18n, groupedRatings, location } = this.props;
+    const { addon, errorHandler, i18n, groupedRatings } = this.props;
     const loading = (!addon || !groupedRatings) && !errorHandler.hasError();
+    const self = this;
+
+    const getLinkTitle = (star, count = '') => {
+      switch (star) {
+        case '5':
+          return i18n.gettext(`Read all ${count} five-star reviews`);
+        case '4':
+          return i18n.gettext(`Read all ${count} four-star reviews`);
+        case '3':
+          return i18n.gettext(`Read all ${count} three-star reviews`);
+        case '2':
+          return i18n.gettext(`Read all ${count} two-star reviews`);
+        case '1':
+          return i18n.gettext(`Read all ${count} one-star reviews`);
+        default:
+          return '';
+      }
+    };
 
-    const linkTitles = {
-      /* eslint-disable quote-props */
-      '5': i18n.gettext('Read all five-star reviews'),
-      '4': i18n.gettext('Read all four-star reviews'),
-      '3': i18n.gettext('Read all three-star reviews'),
-      '2': i18n.gettext('Read all two-star reviews'),
-      '1': i18n.gettext('Read all one-star reviews'),
-      /* eslint-enable quote-props */
+    const createTableRow = (star) => {
+      let starCount;
+      let starCountNode;
+
+      if (!errorHandler.hasError()) {
+        if (groupedRatings) {
+          starCount = groupedRatings[star];
+        }
+
+        starCountNode = i18n.formatNumber(starCount || 0);
+      }
+
+      const rowProps =
+        loading || !addon
+          ? {}
+          : {
+              onClick: () => self.handleRouting(star),
+              title: getLinkTitle(star, starCount),
+            };
+
+      return (
+        <tr className="RatingByStar-table-row" {...rowProps}>
+          <td className="RatingsByStar-star">
+            {!loading ? i18n.formatNumber(star) : <LoadingText width={100} />}
+            <IconStar selected />
+          </td>
+          <td className="RatingsByStar-barContainer">
+            {loading ? (
+              <div className="RatingsByStar-bar RatingsByStar-barFrame" />
+            ) : (
+              <div className="RatingsByStar-bar RatingsByStar-barFrame">
+                {starCount !== undefined
```

Comment:
Can you please check that we have a test for the case when `starCount` is `undefined`.

Reply:
No, we don't have any test for this case. I will add this in the update.

Reply:
I don't think that test is testing what I meant. If `starCount` is `undefined`, then rather than outputting the result of `this.renderBarValue(starCount)`, the code will output `null`, which would mean that there wouldn't even be a `div` with a `className` of `RatingsByStar-barValue`, but the test you mention (`renders 0% bar values when ratings do not exist`) is specifically checking elements with a `className` of `RatingsByStar-barValue`. 

So... either the code isn't working the way it is intended to, or we don't have a test where `starCount` is `undefined`.

Reply:
Yes, actually we don't have test when star count is undefined. I will add that in the next update.